File: 169-majority-element/majority-element.java
class Solution {
    public int majorityElement(int[] nums) {
        int mejority =nums[0];
        int count=1;
 
        for (int i=0;i<nums.length-1;i++){
            if(mejority==nums[i+1]){
                count++;
            }
            else if(mejority!=nums[i+1]){
                count--;
            }
            if (count==0)
            {
                mejority=nums[i];
                count++;
            }
        }
        return mejority;
        
    }
}

File: 169-majority-element/majority-element.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        mejority = nums[0]
        count = 1

        for i in range(1, len(nums)):
            if count == 0:
                mejority = nums[i]
                count = 1
            elif nums[i] == mejority:
                count += 1
            else:
                count -= 1
                
        return mejority 